File: packages/js/core/src/algorithms/find-plugin-package.ts
import { Uri, PluginPackage, PluginRegistration } from "../types";

import { Tracer } from "@polywrap/tracing-js";

export const findPluginPackage = Tracer.traceFunc(
  "core: findPluginPackage",
  (
    uri: Uri,
    plugins: readonly PluginRegistration<Uri>[]
  ): PluginPackage<unknown> | undefined => {
    const pluginRedirect = plugins.find((x) => Uri.equals(x.uri, uri));

    return pluginRedirect?.plugin as PluginPackage<unknown> | undefined;
  }
);

File: packages/py/core/src/algorithms/find_plugin_package.py
from typing import Union, List

from ..types import Uri, PluginPackage, PluginRegistration


def find_plugin_package(uri: Uri, plugins: List[PluginRegistration]) -> Union[PluginPackage, None]:
    plugin_redirect = next(filter(lambda x: Uri.equals(x.uri, uri), plugins), None)
    return plugin_redirect.plugin if plugin_redirect else None
